src/tools/db-upgrade/DbUpgraderTool.cpp

```diff
@@ -67,28 +74,133 @@ void printHelp() {
        --max_concurrent_spaces<N>
          Maximum number of concurrent spaces allowed.
          Default: 5
+
+       --dst_part_file=<file to dataPath and parts>
+         File name of datapaht and part distribution
+         Default: ""
 )");
 }
 
 void printParams() {
   std::cout << "=========================== PARAMS ============================\n";
-  std::cout << "meta server: " << FLAGS_upgrade_meta_server << "\n";
-  std::cout << "source data path: " << FLAGS_src_db_path << "\n";
-  std::cout << "destination data path: " << FLAGS_dst_db_path << "\n";
+  std::cout << "Meta server: " << FLAGS_upgrade_meta_server << "\n";
+  std::cout << "Source data path: " << FLAGS_src_db_path << "\n";
+  std::cout << "Destination data path: " << FLAGS_dst_db_path << "\n";
   std::cout << "The size of the batch written: " << FLAGS_write_batch_num << "\n";
-  std::cout << "raw data from version: " << FLAGS_raw_data_version << "\n";
-  std::cout << "specify all partIds to be processed: " << FLAGS_partIds << "\n ";
+  std::cout << "Raw data from version: " << FLAGS_raw_data_version << "\n";
+  std::cout << "Specify all partIds to be processed: " << FLAGS_partIds << "\n ";
   std::cout << "whether to compact all data: " << (FLAGS_compactions == true ? "true" : "false")
             << "\n";
-  std::cout << "maximum number of concurrent parts allowed:" << FLAGS_max_concurrent_parts << "\n";
-  std::cout << "maximum number of concurrent spaces allowed: " << FLAGS_max_concurrent_spaces
+  std::cout << "Maximum number of concurrent parts allowed:" << FLAGS_max_concurrent_parts << "\n";
+  std::cout << "Maximum number of concurrent spaces allowed: " << FLAGS_max_concurrent_spaces
             << "\n";
+  std::cout << "whether to auto_gen_part_schema: "
+            << (FLAGS_auto_gen_part_schema == true ? "true" : "false") << "\n";
+  std::cout << "File name of datapaht and part distribution is " << FLAGS_dst_part_file << "\n";
+
   std::cout << "=========================== PARAMS ============================\n\n";
 }
 
 using nebula::GraphSpaceID;
 using nebula::PartitionID;
 
+bool upgradeOneDataPathPartSchem(nebula::meta::MetaClient* metaClient,
+                                 GraphSpaceID spaceId,
+                                 const std::string& path,
+                                 std::vector<PartitionID> partIds) {
+  if (!nebula::fs::FileUtils::exist(path)) {
+    LOG(ERROR) << "Data path " << path << " not exists!";
+    return false;
+  }
+
+  auto spaceVidLenRet = metaClient->getSpaceVidLen(spaceId);
+  if (!spaceVidLenRet.ok()) {
+    LOG(ERROR) << "Get space vid len failed, space id " << spaceId;
+    return false;
+  }
+  auto spaceVidLen = spaceVidLenRet.value();
+  std::unique_ptr<nebula::kvstore::RocksEngine> writeEngine(
```

Comment:
I guess the `spaceVidLen` is not actually used? Maybe you could pass a fake one, so we don't need to pass the meta client

Reply:
Yes, it seems that it can only be used when the prefix is used.